presto-orc/src/main/java/com/facebook/presto/orc/stream/InMapOutputStream.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.stream;
+
+import com.facebook.presto.orc.ColumnWriterOptions;
+import com.facebook.presto.orc.DwrfDataEncryptor;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.Optional;
+
+import static com.facebook.presto.orc.metadata.Stream.StreamKind.IN_MAP;
+
+public class InMapOutputStream
+        extends BooleanOutputStream
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(InMapOutputStream.class).instanceSize();
+
+    public InMapOutputStream(ColumnWriterOptions columnWriterOptions, Optional<DwrfDataEncryptor> dwrfEncryptor)
+    {
+        super(columnWriterOptions, dwrfEncryptor);
+    }
+
+    @Override
+    public StreamDataOutput getStreamDataOutput(int column)
+    {
+        // get boolean output DATA stream as IN_MAP stream
+        return super.getStreamDataOutput(column, IN_MAP);
+    }
+
+    @Override
+    public long getRetainedBytes()
+    {
+        return INSTANCE_SIZE + super.getRetainedBytes();
```

Comment:
Instance Size is double counted. If we include InMapOutputStream class size, booleanOutputStream class size need not be included. But this is not a big issue, so this is fine as it is.

Reply:
Indeed, I through that `ClassLayout.parseClass(XYZ.class).instanceSize()` returns only the size of that class w/o counting inheritance. Thanks for pointing it out.

I'll fix it in the follow up PR.

Reply:
I double checked presto-orc, looks like there are two more classes that follow the same incorrect accounting. It's LongDirectSelectiveStreamReader and LongDictionarySelectiveStreamReader.